src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -35,5 +37,39 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::RequestCommissioning(int commissionerNumber)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    const Mdns::DiscoveredNodeData * commissioner;
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        commissioner = GetDiscoveredCommissioner(i);
+        if (commissioner != nullptr && --commissionerNumber == 0)
+        {
+            break;
+        }
+    }
+    if (commissionerNumber != 0)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Enter commissioning mode, open commissioning window
+    InitServer();
```

Comment:
This also does not look right to me.  The "server" should not be controlled by this random component; it's a shared service.  The app should be doing the `InitServer` bits.

Reply:
Moved calls to InitServer, and others to app/server class methods to example app main.cpp